src/hooks/useSubscription.ts

```diff
@@ -45,11 +47,12 @@ export const useSubscription = <T = any, R = T, V = object>(
 
     [unsubscribe.current] = pipe(
       client.executeSubscription(request, devtoolsContext),
-      subscribe(({ data, error }) => {
+      subscribe(({ data, error, extensions }) => {
         setState(s => ({
           fetching: true,
           data: handler !== undefined ? handler(s.data, data) : data,
           error,
+          extensions,
```

Comment:
I don't think we should pass this on to the "user-level" as exchanges would be the only consumers of these special fields.

Or is there any use case you can think of? I'm not sure whether Apollo passes them on?

This can especially confusing because cached results wouldn't have any extensions, which makes this an easy pitfall as well

Reply:
I think this very much could also be part of the data bit fair enough. The problem here though is that we have to assume that the extensions are bound to the specific request.

This means that we shouldn't cache it as it's stale when it's read back from the cache, especially if it contains tracing data and performance data. This probably should also not be included in the SSR result as the frontend shouldn't be concerned about request information from the backend.

I know this is controversial, but I have the feeling that this is very much "non best-practice territory".

If we're thinking of authorisation in general, if it's meta data then a custom exchange would possibly handle that. Otherwise it may be better to include it in data.

A normalised cache would never be able to cache this data, as queries can be fulfilled entirely offline.

So there's definitely a point I believe to surfacing this to the user for "advanced use cases" but the extensions probably shouldn't be cached, shouldn't be part of the SSR serialised data, and shouldn't be guaranteed.

I'm unsure however about some details around this 😅

Reply:
I mean, I'm definitely all for passing extensions on. Yea, that's a nice way of getting some extra data out of exchanges.

But none of the default behaviour should cache or guarantee the extensions. That's definitely custom exchange territory.

So the ssrExchange should probably be modified to remove extensions from the serialised result.

Reply:
> We want to surface extensions in the hooks, we just want to remove my bad assumption re: caching the extensions?

yea, let's surface it. We probably won't put too much focus on it in the docs, but it's particularly useful for also adding (client-only) information/metadata/data from the exchanges, without modifying `data`, like you said.

> I think as well as ssrExchange i need to modify cache.ts to filter extensions before it does a cache set operation.

I think you might need to keep it when you set, because otherwise it may be missing during SSR after suspense is done 🤔  But we definitely want to filter it out from the data that is returned by `serializeData`, so that it's not sent to the client from the server.

Cheers for taking the time to do this btw 🙌

Reply:
Yea, I'll give it a quick review again 👌